src/shaper_srv.erl

```diff
@@ -109,15 +114,15 @@ reset_shapers(ProcName) ->
 init(Args) ->
     State = #state{max_delay = proplists:get_value(max_delay, Args, 3000),
                    ttl = proplists:get_value(ttl, Args, 120),
-                   shapers = dict:new(),
-                   a_times = dict:new()
+                   shapers = #{},
+                   a_times = #{}
```

Comment:
oh, we don't need to store a_times separately.
Shaper has last_update field in the state. Ok, technically it is `last_update + delay`, but should work as well.

We also don't need to store shaper **at all**, if their value is `none` (but we can try to create them each time we want to update them). Even more, we probably could just do `mongoose_config:lookup_opt([shaper, Name])` before we call the shaper_srv and exit fast.


Also, the whole shaper logic could probably be moved to ETS, but that is another topic :D

Reply:
Ok, these comments are out of the scope of this quick PR, I wasn't trying to build better shapers, but just rework generic code to make it faster, if you can reword these details and make it all better, that'd be awesome 🙏🏽